### DISCUSSION TITLE
can I import a list of all installed programs if I know them?

### DISCUSSION THREAD

**Discussion by Ramazottel** [Homebrew/discussions#4424] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.13-28-g4fe3436
ORIGIN: https://github.com/Homebrew/brew
HEAD: 4fe3436682365f81eb68f56a2b8b5d766a372e00
Last commit: 53 minutes ago
Core tap JSON: 14 Apr 21:18 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.87.0 => /usr/bin/curl
macOS: 13.3.1-arm64
CLT: 14.3.0.22.14.1
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: gettext files detected at a system prefix.
These files can cause compilation and link failures, especially if they
are compiled with improper architectures. Consider removing these files:
  /usr/local/lib/libgettextlib.dylib
  /usr/local/lib/libintl.dylib
  /usr/local/include/libintl.h
```

### Description of issue

I just got my new m1 macbook pro 2020 and transferred everything over from my i3 macbook air 2020 to my new mac with migration assistant. As it was done I tried using brew install but it didn't work, so I googled the error. I was able to fix it after a short time but then my old programs that I transferred over with migration assistant didn't show up anymore under brew ls. Is there a way that I can see them again/ export a list of the programs on my old mac and import it into brew on my new mac? I want to see them in brew to update them and stuff like that. I'm sorry for my bad English.

**Comment by p-linnane** [comment]
Homebrew is installed under `/usr/local` on Intel machines, and under `/opt/homebrew` on ARM machines. This is most likely the cause of your issue. Please see @gromgit's instructions in [this post](https://github.com/orgs/Homebrew/discussions/4397#discussioncomment-5567441) for how to resolve this. Follow them carefully.

**Reply by Ramazottel** [reply]
Hey, thank you very much for the reply, all the formulae are there now, but my casks all failed to download again because I already have them. Is there a way to just list the casks in Homebrew or something like that, so that I don't have to delete them all and download/ setup them again and so I'm also able to update them all through Homebrew? I didn't do the last two steps because as of my understanding they're just for cleanup when everything is there.
